gammapy/maps/wcs.py

```diff
@@ -882,6 +882,16 @@ def __repr__(self):
         )
         return str_
 
+    def _check_compatibility(self, other):
+        # check overall shape and axes compatibility
+        if self.data_shape != other.data_shape:
+            raise ValueError("MapGeom data shapes differ")
+        for axis, otheraxis in zip(self.axes, other.axes):
+            axis._check_compatibility(otheraxis)
+
+        # check WCS consistency
+        if self.wcs.wcs.compare(other.wcs.wcs) is False:
```

Comment:
I see that this is exact equality for float numbers:
http://docs.astropy.org/en/stable/api/astropy.wcs.Wcsprm.html#astropy.wcs.Wcsprm.compare

See my other comments: naively I would think that we sometimes want numerical allclose compare, and that we never really need exact float compare, so we could just get away with only implementing the allclose compare?

Reply:
The only issue with `all_close` is that we cannot use the tolerance as a parameter in the arithmetics. So default have values have to be carefully selected.